File: app/app/Feature/Report/Type/ReportType.swift
enum ReportType: String, CaseIterable, Identifiable {

    case abuse
    case spam
    case minor
    case sexual
    case fake
    case etc

    var id: Self { self }

    var title: String {
        switch self {
        case .abuse: return "욕설 / 비방"
        case .spam: return "스팸 / 광고"
        case .minor: return "미성년자"
        case .sexual: return "음란물"
        case .fake: return "도용"
        case .etc: return "기타"
        }
    }
}

File: server/src/main/kotlin/com/pidulgi/server/report/entity/type/ReportType.kt
package com.pidulgi.server.report.entity.type

import com.pidulgi.server.common.exception.CustomException

enum class ReportType(val text: String) {

    ABUSE("욕설 / 비방"),
    SPAM("스팸 / 광고"),
    MINOR("미성년자"),
    SEXUAL("음란물"),
    FAKE("도용"),
    ETC("기타");

    companion object {
        fun from(value: String): ReportType {
            return try {
                ReportType.valueOf(value.uppercase())
            } catch (_: IllegalArgumentException) {
                throw CustomException("신고 유형이 유효하지 않습니다. (입력: $value)")
            }
        }
    }
}